### ISSUE TITLE
Graphics not properly scaled on high dpi screen

### ISSUE THREAD

**Issue by shg** [aviks/GameZero.jl#62]
I tested basic.jl and found that the graphics are only shown in the top-left quadrant of the window on my Mac. I guess high DPI (retina) screen is not supported. The background image is shown correctly maybe because it is a texture that fills the whole surface.

I see some code which may be related to high DPI screen in window.jl but I have no idea how to fix it...

https://user-images.githubusercontent.com/55463/190881745-9c38678d-2b58-4638-a63f-a708a74bf9c1.mov

**Comment by rekabrnalla** [comment]
I also see this problem.

I have to add the *2 to the mouse position to make the cursor properly track.  The window is drawn twice as big as it should be.

`# initialize screen

WIDTH = 600
HEIGHT = 600
BACKGROUND = colorant"antiquewhite"

# define initial state of actors

hline = Line(0, 300, 600, 300)
vline = Line(300, 0, 300, 600)
mouse = Circle(300,300,50)


function draw(g::Game)
    draw(hline, colorant"black")
    draw(vline, colorant"black")
    draw(mouse, colorant"blue", fill=true)
end


#define mouse inputs
 function on_mouse_move(g::Game, pos)
    mouse.x = pos[1]*2
    mouse.y = pos[2]*2
 end
`

**Comment by aviks** [comment]
Yes, unfortunately I haven't been able to find a fix for this yet. 

I use GameZero/SDL2 on Windows with a HiDPI display, and it works correctly. It even works with fractional scaling (my monitors are set to 175% and 150% scaling).

SO suggests the corrent way to handle this in SDL is to set `SDL_WINDOW_ALLOW_HIGHDPI` when creating windows, which we do use [here](https://github.com/aviks/GameZero.jl/blob/fcfd3885ba985863c5148d64dae1102ba13203f0/src/window.jl#L14) , and that seems to work correctly on windows.

Would appreciate any ideas or investigations.

**Comment by rekabrnalla** [comment]
So is there a change to the code above that I need to make to resolve it or is it just unusable on MAC.

We just need to pass this issue to the SimpleDirectMediaLayer people?  Or is there a different combination of switches that MACOS need?
